File: documentation-examples-java/src/java/dev/typr/foundations/docs/landing/OracleTransactor.java
package dev.typr.foundations.docs.landing;

import dev.typr.foundations.Fragment;
import dev.typr.foundations.OracleTypes;
import dev.typr.foundations.RowCodec;
import dev.typr.foundations.Transactor;
import dev.typr.foundations.connect.OracleConfig;

import java.sql.SQLException;

@SuppressWarnings("unused")
public class OracleTransactor {
    //start
    // Oracle - typed config, no JDBC URL to remember
    Transactor tx =
        OracleConfig.builder(
                "localhost", 1521, "xe", "app", "secret")
            .serviceName("XEPDB1")
            .transactor();

    // Everything inside runs in one transaction
    String getGreeting() throws SQLException {
        return Fragment
            .of("SELECT 'Hello from Oracle' FROM dual")
            .query(RowCodec.of(OracleTypes.varchar2)
                .exactlyOne())
            .transact(tx);
    }

    //stop
}

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/landing/OracleTransactor.kt
package dev.typr.foundationskt.docs.landing

import dev.typr.foundationskt.*
import dev.typr.foundationskt.connect.*
import dev.typr.foundationskt.data.*

@Suppress("unused")
class OracleTransactor {
    //start
    // Oracle - typed config, no JDBC URL to remember
    val tx =
        SimpleDataSource.create(
            OracleConfig.builder("localhost", 1521, "xe", "app", "secret")
                .serviceName("XEPDB1")
                .build()
        ).transactor()

    // Everything inside runs in one transaction
    fun getGreeting(): String =
        sql { "SELECT 'Hello from Oracle' FROM dual" }
            .query(RowCodec.of(OracleTypes.varchar2).exactlyOne())
            .transact(tx)

    //stop
}
